### PR TITLE
[Rdy] [CR] Immortal food overhaul - long lasting but rotting food.

### PR DESCRIPTION
**Pull request by nexusmrsep**
This changes lot of food items, that were previuosly 'immortal' (non-perishable) to perishable, long-lasting, but eventualy rotting food.
Also some perishables got prolonged shelf life, equal to their true "state-of-decay" (pun intended).

**Compare:** #22890

**Since spoil time for comestibles is in many cases inbalanced or unstandarized, and I could have just made it worse in few cases, missed some instances, I made it [CR] so every suggestion counts. **

**_[Edit]: during research and discussion many concepts posted initialy changed during this PR progress._**

**Racionale**: Lot of food should eventualy rot, even if it has very long shelf life. Still, some comestibles that we would consider perishable IRL is considered non-perishable for game purposes. While i believe lot of non-perishables like wheat (etc.) should stay this way, others like dry meat, saussages, although cured & salted beyond recognition will not last for more then a ~~year~~ (in a perfect storing conditions).

Translation from real-life lengths of long shelf life food to in-game values can be tedious, when default  year length is 60 days, while a day still lasts full 24h.

In that manner I deliberately used double deafult season length as a reference, since 30 days is (mostly) equal to real-life month.
Season = 30 days, Year = 4 seasons = 30x4=120 days

This way for each previously immortal organic food i asked myself a question: would it last for a week, month, a season, a year, and applied this reference table:
- day = 24 h
- week = 7 d = 168 h
- month = 30 d = 720 h
- season = month = 720 h
- year = 4 seasons = 2880 h

**Some of the results, sorted by duration:**

~~**SHELF-LIFE: MONTH/SEASON**~~ [edit: updated afterwards]
- fluid sac [rationale: the outer plant matter membrane will eventualy rot as being organic]
- sausage, mannwurst,
- hard sinned fruits (and apples as they last long on shelf)
- all aspics
- boiled egg (was too long actualy)

~~**SHELF-LIFE: YEAR**~~ [edit: updated afterwards]
- tallow, tainted tallow, lard
- dehydrated fish, meat & veggies
- toast-em
- pemmican
- pork stick

**OTHER CHANGES:**

**1.** All pickled in a jar, canned, vacuum-packed food has short shelf-life when opened, as opening leaves them prone to outside conditions.

**2.** Apples, ~~pears~~, oranges, lemons, grapefruits, ~~pineaples~~  - those fruits have harder skins and/or natural chemicals in their skins for prolonged rot protection, so - unlike berries, cherries, plums, etc. - ~~I believe~~ they should survive at least one ~~season~~ month of shelf-life (was: 1 week or comparable).
[**edit**: _some of them actualy don't last that long, see comments below]_

**3.** Root vegetables like potatoes, beets, onion have longer shelf life then other veggies. This was not represented properly before.

### CODE REVIEW

**Comment by Rivet-the-Zombie** [comment]
Things like dehydrated foods and pemmican have a shelf life that's measured in _decades_ if they're stored in a dry, cool environment.

File: data/json/items/comestibles.json [hunk 1]
```diff
@@ -3355,10 +3364,11 @@
         "name" : "vegetable aspic",
         "weight" : 120,
         "color" : "green",
+        "spoils_in" : 720,
         "comestible_type" : "FOOD",
         "symbol" : "%",
         "nutrition" : 20,
-        "description" : "A dish in which vegetables are set into a gelatin made from a plant stock.  This food will never spoil.",
+        "description" : "A dish in which vegetables are set into a gelatin made from a plant stock.  This food will last long.",
```
Comment (Cyrano7) [inline data/json/items/comestibles.json]: "This food will last a long time." sounds better.

**Comment by DracoGriffin** [comment]
I'd highly suggest having a source on durations (you can scale them based on that information) rather than deciding arbitrarily.

**Comment by smolbird** [comment]
Agreed. Source your shelf life info or any realism change will be arbitrary and fail at that purpose.

File: data/json/items/comestibles.json (hunk 1 above)
Comment (nexusmrsep) [reply data/json/items/comestibles.json]: Agreed. Will implement shortly.

**Comment by nexusmrsep** [comment]
I also agree, that's a good idea overall. I will provide results shortly.

Still there are few concerns that come to my mind - after comparing few sources its seems that shelf life provided vary - sometimes greatly. The general trend seems to be - the more perishable, short-lived products the more reliable (consistent) info. When you go to potentialy long lasting products many sites give inconsistent results. Some (like preppers sites and company sites) sometimes seem to be biased upon their own beliefs or particular interests. Especialy when they say of nearly "immortal" food lasting "decades" or "indefinite ammounts of time". Unless there will be truly scientific info on such cases i will assume worst case scenario, of non-perfect storing conditions. I will also provide explanation - synthesis of believable information, for further discussion.

I will also try to update the PR on the go, with all research information I get and present here.

This however leads to gameplay question:

**1. Where is the line between perishable and non perishable food game-wise?**
Since this is a game and a simulation, it does not represent life in 100%. Thus it is possible to put an agreeable standard on how far do we go. Let's consider canned food - currently its non-perishable (and I personaly think its a good thing) - while IRL it would have shelf-life of 2-5 years max. Same with items like pasta - in absolutly perfect conditions it would last IRL for 2 years. [https://www.eatbydate.com/grains/pasta-shelf-life-expiration-date/](url)
We may go far: give most food spoil time.
We may go low: set amount of time above which food becomes immortal for game purposes.

**My proposition:**
Golden rule: when in doubt, if food lasts > 2 years IRL, it lasts forever in-game.
- every processed food with packaging lasts forever in-game, as it is preserved and cut off from external conditions (oxygen, water and humidity, bacteria, light)  and when considering shelf life after being opened apply golden rule to determine time of rot
- every processed food without packaging (open air access, prone to humidity, bacteria, light and other conditions not tracked by game, so excluding temperature) when its' IRL shelf life is above 2 years lasts forever in-game
- every processed food without packaging (conditions as above), when its IRL shelf life is below 2 years is perishable and is given IRL spoil rate, with guidelines for month/season conversion as mentioned earlier.

**2. How do we define storing conditions for relevancy?**
In some cases shelf-life conditions presented by most sources is unrealistic in a decrepit wasteland of Cataclysm, not to mention poor hygine (floor pasta here I come), lack of running water and some times proper dishes. Packaging is not implemented (you find packed food but can't pack food). Therefore, unless stated otherwise i will always think as dealing with **non-ideal storing conditions** for food, but I will try not to overdo it. Explanantion will be given. And of course this will be for room temperature.

**Comment by Podesta** [comment]
I would advise strong caution as well. I think there is good work that can be done on making the expiration dates more consistent across the board. But I don't think every food should have a max expiration date of 1 year.

The example of the dried pasta you used is a good one. While it come printed on the pakage as an expiration date or best before around 2 years, it doen't mean it's going to spoil after that. Very far from it. Dried pasta will last much, much longer than that. On long shelf life foods, all that date says is that the manufacturer guarantees the taste will still be the same throughout that period. Plenty of dried/powdered food will last for a very, very long time. Frozen food will never ever spoil. If an agency recommends that you don't store a frozen meat for longer than 6 months is because they think that the taste might not be quite the same after this period. Nothing about it spoiling and you getting sick from eating it.

While our society has an excess of food, with obesity being a real problem, and it only taking a couple minutes to get our hands on whatever we feel like eating, has afforded us to be extremely picky on what we eat, I imagine in a apocalyptic world with survival at stake we would be far less exigent. Maybe giving a small morale penalty as it doesn't taste like what you are used to, but some food shouldn't spoil in the time span of the game.

**Comment by kevingranade** [comment]
I agree with the focus on placing foods into a reasonable "bucket" instead of a precise time.  Possibly (if this is beyond you no problem) we should specify rot time this way instead of with a precise number in the first place.

I'm very surprised at *any* fresh fruits having a shelf life of a month, so you have any sources for that?

Re: Ideal storage, this is a pretty serious issue, and will certainly be a point of misunderstanding if we don't address it.  Many preserved foods are subject to this, dried foods are only non-perishable as long as they remain dry for example.

I'm inclined to treat *all* storage as non-ideal at the moment, and require specific constructions (that can either spawn in mapgen, i.e. a prepper basement or be constructed by the player) in order to treat storage as ideal and make the contained food into permafood.
For example, 99.9% of the pasta you encounter in-game is going to be subject to rat or insect consumption/spoilage, because the grocery stores where it is stored will no longer be protected from these pests.  For the rare cases where it is stored properly, we can add storage constructions that mark it as safe. Fir another example, if we have e.g. beef jerky prepared by the survivor, if stored in a root cellar (especially in a salting barrel) it can last an extremely long time, OTOH if its rattling around a survivor backpack, potentially exposed to not just humidity but direct moisture from rain, it's not going to last nearly as long.

> every processed food with packaging lasts forever in-game, as it is preserved and cut off from external conditions

Agreed, these might become unpalatable at some point, but the whole point is their robustness even poor storage conditions.

> every processed food without packaging when its' IRL shelf life is above 2 years lasts forever in-game

I disagree, in fact I'd assert that few foods without robustly sealed packaging are going to last even a year, because that shelf life assumes protection from pests and humidity extremes.  We can certainly look for counterexamples, but I'd say cap unpackaged foods at a year outside of specially marked "proper storage".

**Comment by DoctorGoat** [comment]
We definitely need fruitcake as a nonspoiling item, as it is fairly trivial to make in a way that keeps it edible for a century if you can make it at all.

**Comment by kevingranade** [comment]
Again, that's less how it's made and more how its stored, alchohol is pretty volatile after all.

**Comment by Zireael07** [comment]
Fresh fruits shelf life of a month is probably (since I find it extremely unlikely otherwise) in refrigerated conditions, as is the case in shopping malls or your own fridge. Therefore doesn't apply in Cata.

On holidays without a real fridge (yachting), we tend to eat fruits within 2 days of getting them. Any longer and we risk them spoiling.

**Comment by nexusmrsep** [comment]
I agree on fresh fruit. Prolonging their life was my mistake and misunderstanding. Their shelf life will be first thing i will either revert in this PR or apply periods based on avalaible sources. Hopefuly later in the day.

**Comment by Podesta** [comment]
> I disagree, in fact I'd assert that few foods without robustly sealed packaging are going to last even a year, because that shelf life assumes protection from pests and humidity extremes. We can certainly look for counterexamples, but I'd say cap unpackaged foods at a year outside of specially marked "proper storage".

I feel like there are some very big gameplay consequences here that should be considered. I imagine robustly sealed would be only glass containers. Limiting all other food to be spoiled after one year, would make exploring for food after this time very limited.

I do like the idea of adding a small chance of all food items to be spoiled whenever they are outside of the reality bubble. They would all be susceptible to various pests and insects, zombies banging on them, blood, leaking roof, etc..

But I think it is a fair assumption that whenever the player is around them (or built a specific structure for them) he is looking after his food storages, and wouldn't be affected in the same way. Could also add a slight preserving factor to lockers/crates/cupboards, so that food placed in them would last longer than food thrown around in the floor.

**Comment by kevingranade** [comment]
> I feel like there are some very big gameplay consequences here that should be considered. I imagine robustly sealed would be only glass containers.

Class containers and canned food, that's a LOT.

> Limiting all other food to be spoiled after one year, would make exploring for food after this time very limited.

That's working as intended.  It's currently too easy.

> But I think it is a fair assumption that whenever the player is around them (or built a specific structure for them) he is looking after his food storages, and wouldn't be affected in the same way.

I disagree, because the player doesn't necessarily have the resources to do so.  Insects and rodents don't avoid food sources based on the mere presence of humans, and temperature and humidity controls are very unlikely to be available to the player in general.  
This is why I said stable storage could be constructable, i.e. a root cellar or similar.

**Comment by nexusmrsep** [comment]
Regarding:

> I'm inclined to treat all storage as non-ideal at the moment, and require specific constructions (that can either spawn in mapgen, i.e. a prepper basement or be constructed by the player) in order to treat storage as ideal and make the contained food into permafood.

and:

> I disagree, in fact I'd assert that few foods without robustly sealed packaging are going to last even a year, because that shelf life assumes protection from pests and humidity extremes. We can certainly look for counterexamples, but I'd say cap unpackaged foods at a year outside of specially marked "proper storage".

I like the idea in general and instantly have few ideas for implemenation to discuss:

In medieval times, and sometimes even nowadays in my country people would use dugouts (root cellars) for food storage. Also consider wine cellars as a different example of the same principle. They would work like natural refrigerators, as earth provides quite stable temeperature. In a a cave average temperature is about 6 degrees Celsius, and in a dugout (shelter) it would be from forementioned temeperature of 6 C to max 10-12 C (around 52 F) all around the year with little flunctuations. It keeps food from freezing (in fact some food hates to be freezed) and in low temperature it would greatly prolong shelf life of many produce (and processed food too) alowing if to survive winter, and heat of the summer. [Side note: people would use cosed buckets with food and dump food in cold water in wells and streams for same reason for short-term storage.]

Here is a very good article on its purpose: [https://en.wikipedia.org/wiki/Root_cellar]
Compare: [here](https://books.google.pl/books?isbn=1429014024) Page 101-102.
and [here](https://www.motherearthnews.com/real-food/root-cellaring/fundamentals-of-root-cellaring-zm0z91zsie). Other good source: [here](http://www.naturallivingideas.com/root-cellar/)
There is much more explanation in those articles then i can provide, so read them out.

As for in-game implementations:

1. Adding a **root cellar // dugout** that would be constructed from the construction menu with: deep pit, 40 stone/bricks (to support walls & humidity control), four 2x4 or 4 long sticks (roof) & (maybe?) 10 dried plants (roof). Loose earth is not a thing, but we can asume there is enough of it from the pit excavation to cover the roof of the cellar.
**Skills**: survival 5 (?), fabrication 2 (?)
**Tools**: tool with digging 2
**Purpose**: every food item placed in square with root cellar // dugout would be treated as having temperature of 6 degrees Celsius, thus it's deterioration process would be significantly reduced by rot time despite outside temperature. Keep in mind that even in winter it would be 6C.
**Also:** great for "alone in the woods" scenarios, with no cities,
**Problem**: some food doesn't like high humidity

2. Basements (z-levels <0 ) should have their temperature regulated by a function. Haven't found one other then one controling ice_lab temperature regresion. General knowledge is that first underground level is like a root cellar above (maybe few degrees less), and the deeper you go the hotter the earth becomes. See mining operations for reference.
Food stored in basements would be somewhat protected by temperature reduction.

2. Mechanics-wise: If rot controlling functions and calculators would determine the environment of the food, and indentify if the square where it is placed has appropriate furniture (or other objects intended for mass food storage) it would NOT worsen rot rate, ie. it would assume it is stored properly.
Problems: different food require different conditions, some food like dry places like attics (beans, flour, corn, wheat, and are less temperature dependant and more humidity dependant.).
Else, if food is stored outside such furniture (on table, on ground, etc.) rot calculator would apply a penalty for ex. x2/x3 rot for a relevant ammount of time.

3. Player should be informed better about food deterioration, and proper storage, otherwise it would become problematic to apply. If given only static info on time-to-rot in perfect conditions, without means to see how much time will it take to rot from present moment in time. I believe it is possible to give this info while examining an item, and it could be related to survival and/or cooking skill for better aproximations.


If you combine all three ideas, player would be encouraged to:
- Store food in proper furniture (bye bye floor pasta).
- Store food in colder places (or not for convinience at cost of faster rot).
- Construct proper furniture and objects for food storage.

Also it would mean that we may actualy apply shelf life for **ideal storage in room temperature**, as all modifications (both bonuses and penalties) would be applied by game mechanics.
This would be great to set an uniform base, a basic principle, becouse its seems we sometimes disagree on shelf-live based on our beliefs on conditions it which food will be stored.

**Example:** Lets say we set our "Made in Wasteland" pemmican shelf-life to 2 years on ideal conditions principle (just an example, not a definite value). Drop that pemican on the ground and leave it in the summer sun and it might last only 2 seasons. Leave it in a locked shelf/drawer in a cupboard and you get your full 2 years. Do the same, but in basement, or place it in a root cellar and here you go - nearly immortal pemmican you always desired.

And all that with food actualy being perishable, which is what this PR is about.
Still as for the scope of this particular PR i only intend to move some non-perishables to perishables and balance some (not all) other comestibles.

Other ideas, if agreed on, would be done in separate PR, and i believe would require more balancing of comestibles afterwards (like maybe moving more non-perishables to perishables, etc.)

**Comment by nexusmrsep** [comment]
Food group: **FRESH FRUIT**
Sources: [ONE](https://visual.ly/community/infographic/food/shelf-life-food), [TWO](http://www.eatbydate.com/fruits/)
Assumptions = time in HOURS, week = 7 days, month = season = 30 days, year = 4 seasons

FOOD | PREVIOUS | PROPOSED | COMMENTS
--------|-----------|---------------|----------------------
apple | 160 | 504 | 3 weeks
banana | 120 | 120 | sources: 2-7 days, game: 5, so unchanged
orange | 96 | 504 | 3 weeks
lemon | 180 | 504 | 3 weeks
pear | 140 | 96 | 4 days
grapefruit | 96 | 336 | 2 weeks
pineapple | 160 | 72 | 3 days
pomegranate | 140 | 336 | 2 weeks, [source](http://www.stilltasty.com/fooditems/index/18037)
kiwi | 120 | 168 | 1 week
// **_to be continued_** //

_changes not yet applied, I will edit this when done_

**Comment by Night-Pryanik** [comment]
In any case, that's weeks, not months.

**Comment by Vasyan2006** [comment]
It will be better to automatically scale long (>1 month IRL) shelf life based on actual seasol length. The same way as it works for seeds growing time.

**Comment by GroeneAppel** [comment]
I must urge the importance of considering season lenght when applying these new rot times. Many players dislike playing with the default season lenghts, and tend to set these far higher. I play with a 90 (!) day season lenght myself. Farming is atrocious for me and a huge investment of time. It would make little sense to use the same rotting system, no matter the season lenght. 

On the aspect of shelf life: A common misconception is the 'recommended' shelf compared to the real shelf life of a product. Take flour for example. The 'best by date' is around 12 months. That however, is the best by date and not expiration date. This does not mean flour isn't edible afterwards, on the contrary, flour can be kept for much longer in decent storage. It simply won't be as tasty afterwards. 
Instead of making such food rot entirely, I would suggest making it 'old' after its best by date, and lower the quality of products made from it. 

As we are discussing the rot of products, I believe this is also a good moment to consider the concept of fridges and ice boxes/freezers. Currently both the mini-fridge and ice-box are part of the game. Their value and function should be considered. Pits used to store have been mentioned, but fridges have certainly not gone unused in the current wasteland. It's not all that hard to keep one running with a battery and some solars. 

Overall, the point of my post is to urge care when making these changes. There is a working system in place, and drastic changes could work out poorly. Especially if no other measures of preservation are added beforehand.

**Comment by smolbird** [comment]
> It will be better to automatically scale long (>1 month IRL) shelf life based on actual seasol length. The same way as it works for seeds growing time.

I disagree. Same reason vitamin accumulation should not scale to the seasons, player activity is on a fixed time frame which will not always be synced with season-length time scales.

In fact, I'd suggest scaling these based on actual real-world lengths, not based on the assumption of 14-day seasons. All perishable food so far is based on real-world logic, we shouldn't break consistency just because the time scale becomes more long-term.

**Comment by GroeneAppel** [comment]
>I disagree. Same reason vitamin accumulation should not scale to the seasons, player activity is on a fixed time frame which will not always be synced with season-length time scales.

I cannot agree with this statement simply due to the fact that right now other mechanics (farming) are already based on season lenght. Either make them all set to a fixed time, or let them all scale. Do not mix them up. 

>In fact, I'd suggest scaling these based on actual real-world lengths, not based on the assumption of 14-day seasons. All perishable food so far is based on real-world logic, we shouldn't break consistency just because the time scale becomes more long-term.

This would create rather amusing situations where players can go for years ingame before certain foods perish. You would have to change the default season lenght drasticly. I can't forsee this working well. 

I suggest scaling because people enjoy playing with different settings. If not scaled automaticly, then offer the players choice in how much they wish to 'scale' the rotting system. Which perhaps would be the superior option above all. It is a sandbox game after all, there is no need to force certain choices down somoenes throat.

**Comment by smolbird** [comment]
> If not scaled automaticly, then offer the players choice in how much they wish to 'scale' the rotting system. Which perhaps would be the superior option above all. It is a sandbox game after all, there is no need to force certain choices down somoenes throat.

Since we can scale construction like this, maybe optional rot scaling might be best, but "use real-world logic" should likely be the default in any case where it's feasible to do so.

**Comment by nexusmrsep** [comment]
The problem with scaling discussed by @Vasyan2006 , @GroeneAppel and @smolbird is not as easy as it seems, becouse of incompatible time scales. You don't change your day length to 48 hours when setting season to 30 days. Nor you set the day to 12 h when choosing standard 15 days season. Day-length (non-compressed-time) is NOT comparable to season length (which for gameplay is compressed-time).

For this reason there is no simple answer to that.

1. If you use 24h scale, then you break the logic for months/seasons. Result: food lasting for many in-game years, while it has for example 356 days on true scale.
2. If you scale for month/seasons (in proportion (ratio) to life), some food would last minutes, while day would be 24h still.
3. If you use @Vasyan2006 idea, it looks good on the outside, but I think in fact it is truly confusing and urcertain. What ratio would you use to convert between those time scales? What would be the base ratio for food rot for this system?

That said I understand that current system which is as @smolbird said based on RL 24h/day scale is not perfect. It's main flaw is based on the same principle. I cannot set yearly rot as 356 days multiplied by 24 hours, for reason stated in point 1. Thus it forces me to propose 30 days/ season as a compromise. Why 30 days? Becuse its as close to 1 IRL monts as can be, and its much easier to deterimine true food shelf life as lots of it falls in this ammount of time. Above one month/season i cannot ignore the fact of shortened year, so i simply assume that 1 year = 4 seasons (months).

It is as rational for a game/simulation simplification as pure fact that it was decided that years will be drasticly shortened in the first place.

Is there a way out from this siutuation that everyone will be satisfied without breaking the game logic? I don't know. If not i prefer @smolbird point of view better.

Regarding:
> Overall, the point of my post is to urge care when making these changes. There is a working system in place, and drastic changes could work out poorly. Especially if no other measures of preservation are added beforehand.
This not the original scope of this PR (making some food non-perishable as is should be from the begining) , but discussion went far beyond that and its good for the game.

Regarding:
> On the aspect of shelf life: A common misconception is the 'recommended' shelf compared to the real shelf life of a product. 
Never have i said that "use by date" or any other factory set date makes any matter in determing true shelf life.

Also a side note: have you tried cooking year-old pasta or baking from year-old flour? You may be surprised that it is not as cooking worthy as it looks, even if eventualy edible. Also sense of taste (same as sense of smell) is not something trivial and for anyone's satisfaction, but primarly to detrmine if consumed food is edible, so loss of taste is a result of loss of food quality, nutrition, edibility etc. If it tastes off would you eat it? If desperate - maybe, if not you would ask yourself twice if it's edible before putting it in your mouth. Eat rancid food and you kick the bucket sooner then later. So yes, you have a small chem lab in your mouth (and nose), and you use it for it's purpouse.

**Comment by GroeneAppel** [comment]
>Is there a way out from this siutuation that everyone will be satisfied without breaking the game logic? I don't know. If not i prefer @smolbird point of view better.

I'm not entirely sure if I understand the way rotting currently works. Does the game currently not track a total game time? With scaling I implied a multiplier which when added to the rotting calculation would increase (or decrease!) the duration food would last. From your explanation however I get the impression that this is not the case and any form of multiplier would be close to impossible to implement. 

>Never have i said that "use by date" or any other factory set date makes any matter in determing true shelf life.

Yes of course, however I held a worry that this would not be taken into consideration in further discussion, which is why I brought it up. Related to your sidenote I have to agree that old food is rather unpleasant to consume. However my point was more focussed on the aspect that many types of food, which once past their recommended by date, typically will last much longer before actually becoming unhealthy to eat. The quality of taste typically will degrade, but the food will often be perfectly safe to eat. I consider any food that causes you to hit the bucket to be spoiled.

Occasionly one can already notice the 'old' status food can get. I was wondering if that could be expanded on. But as you have already said, that is a rather different subject and not suited for this PR.

**Comment by kevingranade** [comment]
> Still as for the scope of this particular PR i only intend to move some non-perishables to perishables and balance some (not all) other comestibles.

Please stick with this, just adjust effective spoilage times assuming that storage is never ideal.  Better storage mechanisms can be added later.

As for what value should go in the json, that should be determined by what can most conveniently derived from irl sources, so if your sources say, "this will be good for one year", then put a number related to that one year figure in the json (your "rounding" scheme is fine, just don't do something like, "source says one year, but since durations are non-ideal that becomes one season").

Do not scale anything based on season length. The only reason plant growth is scaled is to prevent the case where its impossible to grow a crop in a single season.  I'd recommend taking your actual figure and translating it to days, and using that figure, as it's the longest time period where dda agrees with reality.

Dugouts are great as the simplest food storage location, that's exactly the kind of thing I was thinking of.  I also agree that basements are going to have similar benefits.

Side note... many New England basements are under the water table at least occasionally and will flood periodically, so that's fun.

> I think it's safe to assume that Cataclysm products are survivalist's sausages, not those short lived ones.

If there is one set of sausages I agree, though there is certainly room for adding shorter lives sausages later (They're generally tastier and healthier).

> This would create rather amusing situations where players can go for years ingame before certain foods perish.

Amusing and broken are two different things, if you want consistent durations, set your season length to something that matches reality.

**Comment by nexusmrsep** [comment]
> Do not scale anything based on season length. The only reason plant growth is scaled is to prevent the case where its impossible to grow a crop in a single season. I'd recommend taking your actual figure and translating it to days, and using that figure, as it's the longest time period where dda agrees with reality.

Ok, so for this purpose I will use rounded year length of 360 days = 30 days/month x 12 months/year
That equals: 360 days x 24h/day = 8640 hours.

> If there is one set of sausages I agree, though there is certainly room for adding shorter lives sausages later (They're generally tastier and healthier).

Since there is a quite recent addition of sweet sausages (#16311) I believe they are [those](https://en.wikipedia.org/wiki/Italian_sausage), i will consider them standard short-lived non-dry sausage. Other type, also recently added, and wasteland sausages (#22392) will get slightly longer storage time in compare to normal dry sausage, for the reason of how are they are made (wasteland variation is a saltier, hard-core version).

**Comment by nexusmrsep** [comment]
All done, all described changes applied.

**Comment by Granitecosmos** [comment]
It's surprising how little info I can find regarding aspic shelf-life on the internet. Nevertheless, [this](https://www.motherearthnews.com/real-food/food-preservation/how-to-preserve-tomatoes-ze0z1305zsie) mentions 3 weeks while refridgerated but not necessarily sealed airtight. However, I have personal experience with the stuff and can safely state it spoils after about a week if stored at around 10 °C temperature in a cold but not really dry room, if it's basically made as a slightly modified meat/bone soup (and it doesn't melt at room temperature if done right). It indeed spoils in about 3 days at room temperature. Hope this helps you @nexusmrsep I couldn't find any other sources with a quick google search.

**Comment by nexusmrsep** [comment]
Thank you @Granitecosmos for your research. It confirms my knowledge on that mater and supports proposed time of 80 hours, as stated above.

**Comment by FulcrumA** [comment]
It would be good to have such changes complimented with more simple real-world ways of preserving food. For example, just burying eggs in salt - a XVIII or XIX century method - was reported to let them last several times as long, up to maximum of two years of edibility, even if not freshness.

**Comment by nexusmrsep** [comment]
Do you have a source on that? Also: one of abovementioned preservation method PR has been just merged. I plan adding root cellars PR next, so yes, there will be options. Next I will try to approach "proper storage" mentioned in the discusion.

**Comment by Brambor** [comment]
Well shit, now my mountains of pickled veggie and meat will root,
well shit, since short time ago, rotting food spawns roaches.

Otherwise, yeah, like that, changing ALL immortal food is great.
Essentially debuff of _glas jar_ (I already got up to 7!!)

**Comment by FulcrumA** [comment]
@Brambor 
May or may not be balanced by introduction of new methods of preserving food, though.

**Comment by nexusmrsep** [comment]
@Brambor 
Does it rot in a sealed jar or _after_ opening?
Also: keep extra food stocked in a basement (#24073), later (after #24083 gets deployed) use root cellar.

**Comment by FulcrumA** [comment]
I played a bit more after addition of this and some of the values seem really off. General feedback:

@nexusmrsep 
Did you look at the values of just loose portion of produce exposed to elements and temperatures shift? Because for example regular rice, even just stored in a regular bag in a pantry can last theoretically indefinitely - I ate rice which was stored in my non-AC apartment for a couple of years (bought a big bag of it to conserve money) and it suffered no noticeable drop in quality, yet alone turned unedible, myself. And cursory google search seems to prove it was not a singular phenomenon.

There seem to be more examples of edibles like that which both IRL and in game usually come in some packaging rather than just being spread around, in contact with water etc - and I reckon they'd urgently require adjustment for this overhaul to perform its role well.

There's also now a problem of difference between self-made and commercial products. While homemade jerky will expire at some point, higher quality commercial brands in unopened packages last year or two in regular apartment conditions before suffering considerable quality loss (but supposedly still being edible). Properly packaged chocolate is also edible quite a long time after manufacture, though with time and change in temperature its color may change. One of the reasons why chocolate treats were included in old-school MREs even in distant past, where processing to allow said long-shelf life was more primitive.

It may be important to create duplicate entries for similar edibles, the way it is currently with some junk food, for example (different types of toast-em stuff comes to mind). Perhaps making packaging actually offering modifier to shelf life would also help (loose pile of rice vs some in a carboard box or a bag, where the former may perhaps spoil after very long time but the latter should keep indefinitely) and expanding vacuum-packaging as a thing that can be applied to any edible (in certain boundaries of volume) could be of help here as well.

**Comment by nexusmrsep** [comment]
1. As I said, some sources are sometimes contradictory or biased, or not based on scientific results, based on storing conditions that may vary, or based on extreme examples like "that guy ate 25-yo pemmican and lived so pemmican is a super-food for all of you uber survivors and it wil last forever." Knowing that we must use sources, and don't assume that something that happend to you and me is either comparable, and foremose statisticaly representative for the whole. I cooked an 1-yo pack of rice once and it was so hard, that i was considering using a hammer. How is that experience relatable to all this? Answer is - it is not relatable at all. There is an average however and it must be applied. There is no "can last theoretically indefinitely".

2. Packaging. I definitly agree that this is a field that can be worked on. Problem nr 1: You can unpack food, but cannot pack/repack it. Same goes for other items, but for food its most obvious. Consider those few kilograms of flour and salt you drove unpacked in a bicycle wire basket. Funny story, but much less realistic that I'd like. When someone gets this done and out of the way, then everything you say can be considered. Take glas jars and tin cans - they preserve food forever when sealed. Its intended, and fine in my opinion. Rot calculation can include the fact that item is packed. This can be done, but I guess it should be postponed untill you can actualy pack food yourself. Also there is a problem with that: should it apply a bonus or rather should the bare food get a penalty for not being packed?

I will (shortly) create an issue to discuss further food-related expansions ideas that could be implemented.

**Comment by FulcrumA** [comment]
1. Good point. However, in case fo certain foods I've mentioned, when it is a common practice to use them stored for particular length of time in some area, that can be taken into consideration - it may be considered an anecdotal evidence but works as good as any other source that is not a repeated scientific experiment. Both rice and hardtack were stored for long trips and military campaigns in various areas during various periods of time because they could a very long time (years), for example. Use of chocolate-based treats in MRE is also based upon "official" data rather than whims (I'd assume, I don't ever any army created MRE of particular composition without ensuring its viability) and even nowadays on many commercial products an expiration date may serve as a suggestion on how long the product is considered fresh.

Though in case of MRE I have to point out its kinda the case now in game as well, since due to how MRE work (an unrottable item that just gets disassembled into various other items) whatever is inside doesn't rot currently either.

2. I guess some of it could be done either by making more recipes (which, as I mentioned, is not a good choice) or by trying to copy mechanism for current storage of liquids. Whether package would offer a bonus or unpacked food would rot at penalty I don't think is an important issue - either works as in practice, it still means that food outside of the package will spoil faster than inside of it.

Looking forward to a ticket you'll make (if you didn't already) soon to discuss it more in-detail.

**Comment by kevingranade** [comment]
Please re-read the previous conversation.  There is a huge difference between "this food stayed edible in my temperature-controlled and vermin-free home for x amount of time" and, "how long will this food remain edible when stored in a building that has been opened to the environment and is no longer protected from vermin?".

In particular, "regular apartment conditions" no longer exist.  Very nearly all buildings where you can find packaged food are going to be subject to wild temperature and humidity swings, and are subject to vermin infestation.

**Comment by FulcrumA** [comment]
True, but in regards to that do notice the point I am making is also mentioning long voyages and military campaigns where the environment by no means was just "regular apartment conditions". For example, hard-baked (undergoing the process multiple times to minimize water content) hardtack from the time of American Civil War [is](https://youtu.be/Ga5JrN9DrVI?t=57) [edible](http://www.reach-unlimited.com/p/336865029/soldier-food-that-keeps-for-50-years--hardtack-as-survival-rations) nowadays, and while period's MRE-equivalent hardtack was usually wrapped in something to limit access from worms, the packaging wasn't anything particular regular household during time period of the game shouldn't have to routinely store food. 

Which is in big part the point of the comment - to underline the importance of packaging and the difference between homemade and commercially distributed products, where "decent" wrapping (even home-improvised one, not necessarily like the one many commercial products would come in before cataclysm) extends the shelf-life of food extensively, despite non-regulated environment (aforementioned by you, post-cataclysm abandoned buildings) and vermins (which may ignore some types of packaging or even its previously processed content, at least as long as its not directly exposed to the atmosphere due to said packaging) and the food that is treated (using preservatives) and wrapped commercially goes even beyond that - a feature and distinction that is quite important in the context of this ticket.

**Comment by FulcrumA** [comment]
True, but in regards to that do notice the point I am making is also mentioning long voyages and military campaigns where the environment by no means was just "regular apartment conditions". For example, hard-baked (undergoing the process multiple times to minimize water content) hardtack from the time of American Civil War [is](https://youtu.be/Ga5JrN9DrVI?t=57) [edible](http://www.reach-unlimited.com/p/336865029/soldier-food-that-keeps-for-50-years--hardtack-as-survival-rations) nowadays, and while period's MRE-equivalent hardtack was usually wrapped in something to limit access from worms, the packaging wasn't anything particular regular household during time period of the game shouldn't have to routinely store food. 

Which is in big part the point of the comment - to underline the importance of packaging and the difference between homemade and commercially distributed products, where "decent" wrapping (even home-improvised one, not necessarily like the one many commercial products would come in before cataclysm) extends the shelf-life of food extensively, despite non-regulated environment (aforementioned by you, post-cataclysm abandoned buildings) and vermins (which may ignore some types of packaging or even its previously processed content, at least as long as its not directly exposed to the atmosphere due to said packaging) and the food that is treated (using preservatives) and wrapped commercially goes even beyond that - a feature and distinction that is quite important in the context of this ticket.

**Comment by Tharn** [comment]
Mostly good, but don't agree with all of this. Dried pasta, rice, flour and plain old air-cured or heat-cured foodstuffs like dried mushrooms should last a decade or more. They don't need ideal conditions. All they need is a dry, properly sealed container... and you can't tell me that with everything at the player's disposal, he isn't able to create doors that seal properly or scavenge boxes that are airtight. Not everything is broken. The player can find flimsy sealed plastic bags but no intact tupperware, ziplocs, rubber bands or tin foil. See where I'm going?

**Comment by nexusmrsep** [comment]
Sailors from past centuries would disagree. They preserved even water (see: [grog](https://en.wikipedia.org/wiki/Grog)) and even tightly locked barrels of dry products (hardtack, wheat) after 'only' a few months of sailing would develop an unpleasent addition of protein in form of maggots and other insects.

**Comment by nexusmrsep** [comment]
Please move this discussion to #24145.
This PR is merged and its not a good place to discuss any further development ideas.

**Pull request merged**